Update documents for the 3.3.0 release (#2431)
GMLC-TDC/HELICS: CHANGELOG.md

- Tested support for boost 1.80 and CMake 3.24
- Added a [command](https://docs.helics.org/en/latest/user-guide/advanced_topics/commandInterface.html) interface to set/clear time barriers
- Added a [Query](https://docs.helics.org/en/latest/user-guide/advanced_topics/queries.html) to retrieve current time barriers
- Added an index group property to manipulate the internal id which could have an impact on some unusual cases of ordering and remove a potential source of randomness in the final results of a co-simulation
- Added a [callback Federate](https://docs.helics.org/en/latest/user-guide/advanced_topics/CallbackFederate.html) capability which allows a large number of callback based federates to execute on a single core without direct user calls
- Added some additional [callbacks](https://docs.helics.org/en/latest/user-guide/advanced_topics/callbacks.html) for federates
- Added flags on the webserver to allow much easier configuration to external network interfaces
- Added read only property HELICS_PROPERTY_INT_ITERATION_COUNT to get the current iteration count for a federate

### Removed

- ghc::filesystem, since all minimum compilers have support for std::filesystem available.

> - removed travis.ci related configuration and documentation
> - removed an unused and unexposed method in the Core API to retrieve the current iteration count, use HELICS_PROPERTY_INT_ITERATION_COUNT with getProperty to retrieve the same data.

Reviewer: ```suggestion
- Removed Travis CI related configuration and documentation
- Removed an unused and unexposed method in the Core API to retrieve the current iteration count, use HELICS_PROPERTY_INT_ITERATION_COUNT with getProperty to retrieve the same data.
```

### Fixed

> - Fix an issue using very large iteration counts

Reviewer: ```suggestion
- Fixed an issue with using very large iteration counts
```

- Fixed some potential memory leaks in the test set
- Fixed an issue created by the long name tests with the release generation
- Fixed an issue with mismatching key names for publishers

> - removed additional use of std::async in the tests which was causing sporadic failures in the test execution

Reviewer: ```suggestion
- Removed additional use of std::async in the tests which was causing sporadic failures in the test execution
```

- Fixed an issue that could occur when switching between NO_ITERATIONS and ITERATE_IF_NEEDED resulting in deadlock
- Fixed an issue that resulted in a timeout disconnect potentially not working properly
- Fixed the command interface API in the C++98 interface

### Changed

> - additional use of `std::string_view` in internal callbacks and network operations

Reviewer: ```suggestion
- Additional use of `std::string_view` in internal callbacks and network operations
```

> - cleaned up use of internal flags and separated them by function

Reviewer: ```suggestion
- Cleaned up use of internal flags and separated them by function
```

- Updated the docker image in use for no_zmq and octave tests
- Updated Utilities, Units, ASIO, fmtlib, gtest to latest versions

> - refactor endpoint management code for consistency with other interfaces

Reviewer: ```suggestion
- Refactored endpoint management code for consistency with other interfaces
```

- The HELICS_DATA_TYPE_CHAR is now a member of the enumeration vs a standalone definition there is no change in operation but the numerical value is now different

> - the change detection on inputs/publication now can work individually for each interface

Reviewer: ```suggestion
- The change detection on inputs/publication now can work individually for each interface
```

> - The default webserver ports

Reviewer: ```suggestion
- The default webserver ports have been changed to 43542 for the HTTP server and 43543 for the Websocket server
```

- The REST API now returns a structure on successful broker creation
- The helics_broker executable now has the same command line arguments for the webserver as the broker_server

### Added

- Custom Translator functionality in the C API
- Added an asynchronous time coordinator which could be used for testing, or if all federates are driven by real time mode, or if internal synchronization is not required.
- Added alias operations to allow interfaces to have multiple names
- Added CORS access control options to the web server
- Added a HELICS_STATE_UNKNOWN as a potential return value used when the federate does not exist
- A time gate to the publications and input to restrict publications to a certain period

> - added encryption related options to vcpkg

Reviewer: ```suggestion
- Added encryption related options to vcpkg
```

> - Tested support for boost 1.80 and CMake 3.24

Reviewer: ```suggestion
- Tested support for Boost 1.80 and CMake 3.24
```

- Added a command interface to set/clear time barriers
- Added a Query to retrieve current time barriers
- Added an index group property to manipulate the internal id which could have an impact on some unusual cases of ordering and remove a potential source of randomness in the final results of a co-simulation
- Added a callback Federate capability which allows a large number of callback based federates to execute on a single core without direct user calls
- Added some additional callbacks for federates
- Added flags on the webserver to allow much easier configuration to external network interfaces
- Added read only property HELICS_PROPERTY_INT_ITERATION_COUNT to get the current iteration count for a federate

### Removed

- ghc::filesystem, since all minimum compilers have support for std::filesystem available.
- removed travis.ci related configuration and documentation

> - removed an unused and unexposed method in the Core API to retrieve the current iteration count

Reviewer: ```suggestion
- Removed travis.ci related configuration and documentation
- Removed an unused and unexposed method in the Core API to retrieve the current iteration count
```
